### PR TITLE
Upgrade Vitest to v5

### PR DESCRIPTION
**Pull request by nwalters512**
# Description

Upgrade Vitest and its V8 coverage provider from 4.1.10 to 5.0.1 across the workspace, and deduplicate the lockfile. Update CI artifact upload/download paths and the Git ignore rule for Vitest 5's new `.vitest/blob` report directory.

Also make `@prairielearn/compiled-assets.close()` await disposal of both esbuild contexts. PrairieLearn already calls this during test and server teardown, but the package previously returned before disposal completed. This closes a shutdown race relevant to Vitest 5 waiting for worker output streams to drain. Include a patch changeset for this runtime fix; the test dependency upgrade itself has an empty changeset.

The existing `html-validate` version reports an optional Vitest peer warning; we use its validator API, not its Vitest integration.

- [x] I have disclosed the level of AI assistance used: Codex implemented the changes and performed local validation.

# Testing

Verified blob-report generation and report merging with V8 coverage using the updated artifact directory. Exercised existing tests covering hoisted KMS mocks, AWS IMDS callbacks, UI filter callbacks, and grader queue mocks under Vitest 5.

Verified repeated asset initialization/cleanup under Vitest 5 with a temporary diagnostic that withholds an esbuild heartbeat reply; the worker exited successfully. The existing development and production asset-serving tests also passed.

Database-backed application tests were not run because local Postgres was unavailable.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -103,7 +103,7 @@ packages/*/coverage
 coverage.xml
 
 ## Vitest
-.vitest-reports
+.vitest
```
Comment (jonatanschroeder) [inline .gitignore:106]: Nit: should the previous directory also be ignored to avoid problems with devs that migrate from before and have historical files here?

Comment (nwalters512) [reply .gitignore:106]: Resolved in 4de4202674 - Kept `.vitest-reports` ignored alongside `.vitest` so historical reports and reports generated when switching to older branches stay out of `git status`.

*- Codex*

File: package.json [hunk 2]
```diff
@@ -55,7 +55,7 @@
     "@prairielearn/tsconfig": "workspace:^",
     "@types/node": "^24.13.3",
     "@typescript/native": "npm:typescript@^7.0.2",
-    "@vitest/coverage-v8": "^4.1.10",
+    "@vitest/coverage-v8": "^5.0.0",
```
[the remark this answers is not in the document]
Comment (nwalters512) [reply package.json:58]: > The supplied lockfile context resolves them to 5.0.1, which conflicts with the stated three-day release delay.

You sure about that?

[the remark this answers is not in the document]
Comment (nwalters512) [reply package.json:58]: My dude, check when 5.0.1 was published and what the current date/time is.

**Review by EduardoMVAz — APPROVED** [review]
